### ISSUE TITLE
Slow build process since sh 1.12.5

### ISSUE THREAD

**Issue by bobatsar** [kivy/python-for-android#1038]
Just for you to know, the performance problem when building a dist and recipies are due to a regression in the python module sh since version 1.12.5

I already wrote a bug report and they are working on it, see amoffat/sh#378

**Comment by inclement** [comment]
Thanks for the report, I've heard about this but didn't actually know what caused it. If it takes any time for the sh devs to fix it, we can try to work around it in p4a by avoiding the sh module where there are many similar commands.

**Comment by bobatsar** [comment]
In fact just deactivating the following kwargs removes the problem but of course the live output is not workingwhile a command is running, as sh blocks until the command finishes. But this is already a good workaroud, but I would wait a little bit if it gets fixed soon.


```
kwargs["_iter"] = True
kwargs["_out_bufsize"] = 1
kwargs["_err_to_out"] = True
kwargs["_bg"] = True
```

**Comment by tito** [comment]
In the meantime: ```pip install sh\<1.12.5```
Maybe we could reference it manually in the setup.py

**Comment by bobatsar** [comment]
It seems they fixed it in sh 1.12.14 (see amoffat/sh#378). I didn't have time to test it.

If anyone can test it this issue can be closed.

**Comment by inclement** [comment]
This was worked around on master, no longer an issue.
